build-aux/snap/snapcraft.yaml

```diff
@@ -29,173 +45,723 @@ license: GPL-3.0
 # See the comments from jdstrand in
 # https://forum.snapcraft.io/t/5547/10
 parts:
-  snapd-deb:
+  build-base:
+    plugin: nil
+    source: keyrings
+    build-packages:
+      - to i386:
+        - debootstrap
+        - fakeroot
+      - else:
+        - wget
+      - gawk
+    build-environment:
+      - RELEASE: "22.04.2"
+      - BASE: ubuntu-base-${RELEASE}-base-${CRAFT_TARGET_ARCH}.tar.gz
+      - DIR_URL: https://cdimage.ubuntu.com/ubuntu-base/releases/${RELEASE}/release
+      - URL: ${DIR_URL}/${BASE}
+      - SHA256: ${DIR_URL}/SHA256SUMS
+      - SIG: ${SHA256}.gpg
+      - DEPENDENCIES: >-
+          make
+          gcc
+          binutils
+          gawk
+          python3
+          python3-jinja2
+          meson
+          bison
+          flex
+          ninja-build
+          libc6-dev
+          gperf
+          autoconf
+          automake
+          gettext
+          libtool
+          pkg-config
+          libglib2.0-dev
+          g++
+          autopoint
+          git
+    override-pull: |
+      craftctl default
+      case "${CRAFT_TARGET_ARCH}" in
+        i386)
+        ;;
+        *)
+          wget "${SHA256}" "${SIG}" "${URL}"
+          gpg --no-default-keyring \
+              --keyring ./cd-image-keying.gpg \
```

Comment:
note this keyring is available by default on all ubuntu installs as ubuntu-archive-keyring.gpg.

Similarly cloud-images keyring is available as a package too.